frontend/src/components/AnalyticsTable/AnalyticsTable.tsx

```diff
@@ -0,0 +1,274 @@
+
+  const initRideData = (data: TableData[]) => {
+    const rideData: Cell[][] = [];
+    data
+      .sort((a, b) => (a.year + a.monthday < b.year + b.monthday ? 1 : -1))
+      .forEach((d) => {
+        const month = d.monthday.substring(0, 2);
+        const day = d.monthday.substring(2);
+        const date = `${month}/${day}/${d.year}`;
+        const dailyTotal = d.dayCount + d.nightCount;
+        const rideRow = [
+          date,
+          dailyTotal,
+          d.dayCount,
+          d.dayNoShows,
+          d.dayCancels,
+          d.nightCount,
+          d.nightNoShows,
+          d.nightCancels,
+        ];
+        rideData.push(rideRow);
+      });
+    setRideTableData(rideData);
+  };
+
+  const initDriverData = (data: TableData[]) => {
+    const driverData: Cell[][] = [];
+    data
+      .sort((a, b) => (a.year + a.monthday < b.year + b.monthday ? 1 : -1))
+      .forEach((d) => {
+        const month = d.monthday.substring(0, 2);
+        const day = d.monthday.substring(2);
+        const date = `${month}/${day}/${d.year}`;
+        const dailyTotal = Object.values(d.drivers).reduce((a, b) => a + b, 0);
+        const driverRow = [date, dailyTotal];
+        driverNames.forEach((driver) => {
+          driverRow.push(d.drivers[driver] || 0);
+        });
+        driverData.push(driverRow);
+      });
+    setDriverTableData(driverData);
+  };
+
+  const handleEdit = (
+    rowIndex: number,
+    cellIndex: number,
+    date: string,
+    value: number,
+  ) => {
+    const cols = type === 'ride' ? dbRideCols : dbDriverCols;
+    setEditData((prev) => {
+      const newVal = { ...prev };
+      const dateEdit = newVal.dates[date];
+      const offset = 2;
+      const index = cellIndex - offset;
+      if (type === 'driver') {
+        if (dateEdit === undefined) {
+          // need to populate all drivers
+          const driverRow = driverTableData![rowIndex].slice(offset);
+          const driversEdit: ObjectType = {};
+          driverNames.forEach((name, i) => {
+            driversEdit[name] = driverRow[i];
+          });
+          driversEdit[cols[index]] = value;
+          newVal.dates[date] = {
+            drivers: driversEdit,
+          };
+        } else {
+          newVal.dates[date].drivers = {
+            ...dateEdit.drivers,
+            [cols[index]]: value,
+          };
+        }
+      } else {
+        newVal.dates[date] = {
+          ...dateEdit,
+          [cols[index]]: value,
+        };
+      }
+      return newVal;
+    });
+  };
+
+  const handleSubmit = () => {
+    setEditData({ dates: {} });
+    setIsEditing(false);
+  };
+
+  useEffect(() => console.log(editData), [editData]);
+
+  useEffect(() => {
+    if (type === 'ride') {
+      initRideData(table.data);
+    } else {
+      initDriverData(table.data);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [type, drivers.length]);
+
+  return (
+    <div className={styles.tableContainer}>
+      <button
```

Comment:
This button is not accessible as it has no text describing what it's doing. Add an aria-label to it to make it accessible (see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute)